classifier.js

```diff
@@ -89,48 +91,44 @@ class Classifier {
 
     // Calculate all the frequences
     // word count / doc count
-    for (let i = 0; i < this.wordList.length; i++) {
-      let key = this.wordList[i];
+    this.wordList.forEach(key => {
       let word = this.dict[key];
 
-      for (let j = 0; j < this.categoryList.length; j++) {
-        let category = this.categoryList[j];
-
+      this.categoryList.forEach(category => {
         // If this word has no count for the category set it to 0
         // TODO: better place to do this or unecessary?
         if (word[category] === undefined) {
-          word[category] = {};
-          word[category].count = 0;
+          word[category] = {
+              count: 0
+            };
         }
         // Average frequency per document
-        word[category].freq = word[category].count / this.categories[category].docCount;
-      }
-    }
-
-    for (let i = 0; i < this.wordList.length; i++) {
-      let key = this.wordList[i];
+        let taxonomy = word[category];
+        let cat = this.categories[category];
+        let freq = taxonomy.count / cat.docCount;
+        taxonomy.freq = freq;
+      });
+    });
+
+    this.wordList.forEach(key => {
       let word = this.dict[key];
       // Probability via Bayes rule
-      for (let j = 0; j < this.categoryList.length; j++) {
-        let sum = 0;
-        for (let k = 0; k < this.categoryList.length; k++) {
-          let category = this.categoryList[k];
-          let freq = word[category].freq;
-          if (freq) {
-            sum += word[category].freq;
-          }
-        }
-        let category = this.categoryList[j];
-        word[category].prob = word[category].freq / sum;
-        // Nothing is certain
-        // TODO: Is there a better way to handle this?
-        if (word[category].prob < 0.01) word[category].prob = 0.01;
-        if (word[category].prob > 0.99) word[category].prob = 0.99;
-      }
-    }
+      this.categoryList.forEach(category => {
+        let sum = this.categoryList.reduce((p, cat) => {
+            let freq = word[cat].freq;
+            if (freq) {
+              return p + freq;
+            }
+            return p;
+          }, 0);
+        let taxonomy = word[category];
+        // Constrain the probability
+        let prob = taxonomy.freq / sum;
+        taxonomy.prob = Math.max(0.01, Math.min(0.99, prob));
```

Comment:
keep the TODO notice here, the code is better but I'm unsure of the 0.01 / 0.99 choice.